mtp_cashbook/apps/mtp_auth/models.py

```diff
@@ -19,6 +19,28 @@ def save(self, *args, **kwargs):
     def is_authenticated(self, *args, **kwargs):
         return True
 
+    @property
+    def username(self):
+        return self.user_data.get('username')
+
+    @property
+    def get_full_name(self):
+        if not hasattr(self, '_full_name'):
+            name_parts = [
+                self.user_data.get('first_name'),
+                self.user_data.get('last_name')
+            ]
+            self._full_name = ' '.join(filter(None, name_parts))
+        return self._full_name
+
+    @property
+    def prison(self):
+        try:
+            return self.user_data.get('prisons', [])[0]
```

Comment:
The cashbook doesn't know which prisons the user has access to, so you access this prop to get that data. We support only one prison at the moment so I was thinking to add a warning when authenticating. Something like: `if prisons.length > 1 => log warning`. Not sure if it helps though.